### PR TITLE
feat: Profile View Model Unit Tests

### LINKED ISSUE #49: test: create tests for User Profile viewing and editing feature
## Description
As a developer,
I need create tests for the user profile feature,
so that I can be confident the code is working properly.

### Update [Required]
- [ ] unit tests are available for this feature.
 - [ ] UI tests are available for this feature.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
[INSERT NUMBER HERE] hours

### PR DESCRIPTION
**Pull request by yugantarjain**
### Description
This PR implements unit tests for Profile Relation View Model. The UserDefaults are tested too.

Fixes #49 

### Type of Change:
- Code
- Quality Assurance

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Xcode and Simulator.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged 
**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: mentorship iosTests/ProfileViewModelTests.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        XCTAssertEqual(sampleProfileData, getProfileData)
+    }
+    
+    func testEditProfileDataReceived() {
+        //prepare sample data
+        let sampleData = ProfileModel.ProfileData(id: 100, name: nil, username: "username", email: "test@abc.com")
+        
+        //save sample data to profile
+        //reason: getEditProfileData uses getProfile function to get stored data. Hence it will get this sample data now.
+        profileVM.saveProfile(profile: sampleData)
```
Comment (ApheleiaS) [inline mentorship iosTests/ProfileViewModelTests.swift:42]: Is getEditProfileData a private function? If not call it directly here?

File: mentorship iosTests/ProfileViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        
+        // get edit profile data
+        let editProfileData = profileVM.getEditProfileData()
+        
+        //id and email should be unchanged
+        XCTAssertEqual(editProfileData.id, sampleData.id)
+        XCTAssertEqual(editProfileData.email, sampleData.email)
+        
+        //name should be empty string (""). nil values are converted to allow for safe force unwrapping
+        XCTAssertEqual(editProfileData.name, "")
```
Comment (ApheleiaS) [inline mentorship iosTests/ProfileViewModelTests.swift:52]: You should have multiple samples in such testcases for values which cover: 
success cases, failure cases, edge cases etc. 

For eg: 
Say for name: 
You should test a string value, a numerical only value, nil value etc. 

Not saying you need to here. It completely depends on what kinds of values are allowed or disallowed to be stored by your data variables.

**Review by ApheleiaS — DISMISSED** [review]

**Review by sunjunkie — CHANGES_REQUESTED** [review]
@yugantarjain This looks fine. Can you resolve the conflicts?

File: mentorship iosTests/ProfileViewModelTests.swift (hunk 1 above)
Comment (yugantarjain) [reply mentorship iosTests/ProfileViewModelTests.swift:42]: We do that. Line 45

File: mentorship iosTests/ProfileViewModelTests.swift (hunk 2 above)
Comment (yugantarjain) [reply mentorship iosTests/ProfileViewModelTests.swift:52]: Yeah. I've tried to include different cases while initialising the sample data through different properties (name = nil, email = "email"), etc.
I see you approved these changes earlier, should I update the PR or is it fine?

**Comment by yugantarjain** [comment]
> @yugantarjain This looks fine. Can you resolve the conflicts?

@sunjunkie Done. Thanks!

**Review by sunjunkie — APPROVED** [review]

**Review by ApheleiaS — APPROVED** [review]
